File: src/SIL.Machine.Shared/WebApi/Dtos/WordGraphArcDto.cs
namespace SIL.Machine.WebApi.Dtos
{
	public class WordGraphArcDto
	{
		public int PrevState { get; set; }
		public int NextState { get; set; }
		public float Score { get; set; }
		public string[] Words { get; set; }
		public float[] Confidences { get; set; }
		public RangeDto SourceSegmentRange { get; set; }
		public bool IsUnknown { get; set; }
		public AlignedWordPairDto[] Alignment { get; set; }
	}
}

File: src/SIL.Machine.TS/src/web-api/word-graph-arc-dto.ts
import { AlignedWordPairDto } from './aligned-word-pair-dto';
import { RangeDto } from './range-dto';

export interface WordGraphArcDto {
  prevState: number;
  nextState: number;
  score: number;
  words: string[];
  confidences: number[];
  sourceSegmentRange: RangeDto;
  isUnknown: boolean;
  alignment: AlignedWordPairDto[];
}
